File: Backend/Basic libraries/Basic libraries/Assemblies.lib.vb
Imports System.Reflection
Imports System.Windows.Forms
Imports AeonLabs.Environment

Public Module DLLlibrary

    Public Function GetFormByName(ByVal FormName As String) As Form
        Dim T As Type = Type.GetType(FormName, True, True)
        Return CType(Activator.CreateInstance(T), Form)
    End Function

    Public Function loadExternalAssembly(ByVal state As environmentVarsCore, ByVal dllFileName As String, ByVal className As String) As Object
        Try
            Dim fullPath As String = state.libraryPath & dllFileName
            Dim asm As Assembly = Assembly.LoadFrom(fullPath)
            Dim myType As System.Type = asm.GetType(asm.GetName.Name & "." & className)
            Return Activator.CreateInstance(myType)
        Catch ex As Exception
            Return Nothing
        End Try
        Return Nothing
    End Function

End Module

File: Backend/Basic Libraries/Basic Libraries/Assemblies.lib.cs
using System;
using System.Reflection;
using System.Windows.Forms;

namespace AeonLabs.BasicLibraries
{
    public static class DLLlibrary
    {
        public static Form GetFormByName(string FormName)
        {
            var T = Type.GetType(FormName, true, true);
            return (Form)Activator.CreateInstance(T);
        }

        public static object loadExternalAssembly(environmentVarsCore state, string dllFileName, string className)
        {
            try
            {
                string fullPath = state.libraryPath + dllFileName;
                var asm = Assembly.LoadFrom(fullPath);
                var myType = asm.GetType(asm.GetName().Name + "." + className);
                return Activator.CreateInstance(myType);
            }
            catch (Exception ex)
            {
                return null;
            }

            return null;
        }

        public static FormCustomized loadAssembly()
        {
            // TODO
            try
            {
                var assembly = Assembly.LoadFile("store.dll");
                var type = assembly.GetType("AeonLabs.storeMainForm");
                Form SetupForm = Activator.CreateInstance(type) as Form;
                var TypesOnAssemblies = SetupForm.GetType().GetProperty("TypesOnAssemblies");
                TypesOnAssemblies.SetValue(SetupForm, type);
                var enVarsSetup = SetupForm.GetType().GetProperty("ExternalLoadEnVars");
                enVarsSetup.SetValue(SetupForm, "");
                SetupForm.ShowDialog();
            }
            catch (Exception ex)
            {
            }

            return default;
        }

        public static FormCustomized loadFormFromAssembly(string fileName, string formName, environmentVarsCore enVars, environmentVarsCore.updateMainLayoutDelegate updateMainApp)
        {
            var formToLoad = enVars.assemblies(fileName).Item(formName).assemblyFormToLoad;
            PropertyInfo setEnVars = formToLoad.GetType().GetProperty("envars");
            setEnVars.SetValue(formToLoad, enVars);
            PropertyInfo setUpdateMainApp = formToLoad.GetType().GetProperty("updateMainApp");
            setUpdateMainApp.SetValue(formToLoad, updateMainApp);
            return formToLoad;
        }
    }
}